データベース設計

akirausui19871117 opened the pull request:
#what
データベース設計及びER図の作成

#why
Furimaのエンティティの洗い出し及びデータ同士の関連を明らかにし、各機能を実装するため







https://gyazo.com/668356c390be3187ccd68e2c90faedb6

akirausui19871117 committed:
データベース設計

k-kz commented on README.md:
```diff
@@ -1,24 +1,54 @@
-This README would normally document whatever steps are necessary to get the
-application up and running.
+###users テーブル
+|column    |types |Options    |
+|----------|------|-----------|
+|nickname  |string|null: false|
+|email     |string|null: false|
+|password  |string|null: false|
+|name      |string|null: false|
+|birthday  |string|null: false|
```
**[usersテーブルに関して]**
カラムが不足しています。
再度サンプルのフリマアプリを参考に、他にも必要がないか見直してみましょう。

k-kz commented on README.md:
```diff
@@ -1,24 +1,54 @@
+has many :buys
+
+###items テーブル
+|column    |types         |Options                       |
+|-----------|-------------|------------------------------|
+|item_name  |string       |null: false                   |
+|explanation|text         |null: false                   |
+|price      |integer      |null: false                   |
+|user       |reference    |null: false, foreign_key: true|
```
**[itemsテーブルに関して]**
カラムが不足しています。
再度サンプルのフリマアプリを参考にしてみましょう。

k-kz reviewed (commented):
お待たせしました、レビュー依頼ありがとうございます！
こちらのプルリクエストですが、マークダウン記法が反映されておりません。
今のうちにマスターするため、こちらのプルリクエストに反映させてみましょう。
#の後にスペースを開けることで機能するかと思いますので、ご確認お願いします。
上はスペースを入れており、下に関してはスペースを入れていません。

# マークダウン
#マークダウン

編集後、再度レビュー依頼フォームにてご提出をお願いいたします。

akirausui19871117 committed:
指摘事項1

マークダウンの追記とカラムの追加

akirausui19871117 commented:
ご確認いただきましてありがとうございました！ご指摘の箇所を修正いたしましたので再度御確認頂ますようお願いいたします。

akirausui19871117 commented:
https://gyazo.com/2430248d1224e00778e3936f378b67b3

k-kz commented on README.md:
```diff
@@ -1,24 +1,63 @@
+### buyテーブル
+|column     |types    |Options                       |
+|-----------|---------|------------------------------|
+|items      |reference|null: false, foreign_key: true|
```
```ruby
|items      |reference|null: false, foreign_key: true|
```
`items`を`item`にしておきましょう。

k-kz commented on README.md:
```diff
@@ -1,24 +1,63 @@
+|-----------|---------|------------------------------|
+|items      |reference|null: false, foreign_key: true|
+|user       |reference|null: false, foreign_key: true|
+|destination|reference|null: false, foreign_key: true|
```
```ruby
|destination|reference|null: false, foreign_key: true|
```
detinationに対してはbuyテーブルが親テーブルであるため、こちらの外部キーを保存するカラムは不要となります。

k-kz commented on README.md:
```diff
@@ -1,24 +1,63 @@
+|building    |string |null: false                   |
+|phone_number|string |null: false                   |
+|user        |string |null: false, foreign_key: true|
+
```
[destinationテーブルに関して]
belongs_toでアソシエーションを組んでいるためbuyテーブルの外部キーを保存するカラムを作成しましょう。

k-kz reviewed (commented):
レビュー依頼ありがとうございます！
コメントいたしましたので、ご確認をお願いいたします。

akirausui19871117 committed:
修正その２

１，active_hashの利用を想定し、プルダウンメニューで選択する箇所を単数形＋idへ変更
２，buy及びdestinationテーブルの外部キーの関係を修正
３，アソシエーションの修正

akirausui19871117 commented:
ご指摘いただきましてありがとうございました！修正を加えました。また、ご指摘に該当する他の箇所も修正を加えました。
お手数ですが、再度御確認頂ますようお願いいたします。

https://gyazo.com/e4aee9c58bde19b7d2f0b889a3c64ebe

shodai-ito commented on README.md:
```diff
@@ -1,24 +1,63 @@
+|first_name        |string|null: false            |
+|family_name_kana  |string|null: false            |
+|first_name_kana   |string|null: false            |
+|birthday          |string|null: false            |
```
誕生日のように日付を保存するデータ型としてdateというものがあります。
そちらの方がより適切かと存じますので変更しましょう！

shodai-ito commented on README.md:
```diff
@@ -1,24 +1,63 @@
+|addresses     |string   |null: false                   |
+|building      |string   |null: false                   |
+|phone_number  |string   |null: false                   |
+|user          |reference|null: false, foreign_key: true|
```
今回の実装ではuserテーブルとアソシエーションを組まなくてOKです！
購入するたびに住所を入力する実装であり、入力された住所が誰のものなのかは現時点で判断する必要がないためです！

shodai-ito reviewed (commented):
レビュー依頼ありがとうございます！
修正点確認させていただきました！
２点コメントしましたので、ご確認をお願いいたします！

再度レビュー依頼フォームのご提出をよろしくお願いいたします。

akirausui19871117 commented:
ご指摘いただきましてありがとうございました！
該当箇所を修正いたしましたので、再度御確認頂ますようお願いいたします。

https://gyazo.com/10363cb07da6fe607a7485e575a155b0

akirausui19871117 committed:
修正その３

１，日付データの型指定をdateへ変更
２，destinationテーブルからuserとのアソシエーションを消去

s-aka-moto commented on README.md:
```diff
@@ -1,24 +1,62 @@
+|column            |types |Options                |
+|------------------|------|-----------------------|
+|nickname          |string|null: false            |
+|email             |string|null: false            |
+|encrypted_password|string|null: false,unique:true|
```
unique:trueはencrypted_passwordではなく、emailにつけてあげましょう！

|email             |string|null: false            |
|encrypted_password|string|null: false,`unique:true`|

s-aka-moto commented on README.md:
```diff
@@ -1,24 +1,62 @@
+|column      |types  |Options                       |
+|--------------|---------|------------------------------|
+|postal_code   |string   |null: false                   |
+|prefecture_id |integer  |null: false                   |
```
都道府県のカラムはitemテーブルと同じモデルを使い回すことができるため、こちらも「delivery_area_id」というカラム名にしてあげましょう！(またはどちらも「prefecture_id」としても良いですね)

s-aka-moto commented on README.md:
```diff
@@ -1,24 +1,62 @@
+|--------------|---------|------------------------------|
+|postal_code   |string   |null: false                   |
+|prefecture_id |integer  |null: false                   |
+|addresses     |string   |null: false                   |
```
番地(もしくは市町村)を入れるカラムを追加しましょう！

s-aka-moto commented on README.md:
```diff
@@ -1,24 +1,62 @@
+|postal_code   |string   |null: false                   |
+|prefecture_id |integer  |null: false                   |
+|addresses     |string   |null: false                   |
+|building      |string   |null: false                   |
```
建物名は必須入力ではないため、「null: false 」は削除しておきましょう！
`|building      |string   |null: false                   |`

s-aka-moto reviewed (commented):
レビュー依頼ありがとうございます！
修正点確認させていただきました。
追加でコメントいたしましたので、ご確認お願いします！
あと少しでございます！頑張りましょう！

akirausui19871117 committed:
修正その４

１，市町村のカラム追加
２，delivery_area_id をprefecture_idへ統一
３，unique:trueの位置の修正
４，buildingの部分のnull:falseの削除

akirausui19871117 commented:
ご指摘いただきましてありがとうございました！
再度修正いたしましたので御確認頂ますようお願いいたします

akirausui19871117 commented:
https://gyazo.com/5b2cbaea18e924962c112e28402c0de7

akirausui19871117 committed:
修正その４−１

sgwr-0113 commented on README.md:
```diff
@@ -1,24 +1,63 @@
+|prefecture_id   |integer      |null: false                   |
+|delivery_day_id |integer      |null: false                   |
+|price           |integer      |null: false                   |
+|user            |reference    |null: false, foreign_key: true|
```
```
|user            |reference    |null: false, foreign_key: true|
```

referenceではなく`references`ですね！
また、**references型にNotNull制約に関する言及は不要**です。

sgwr-0113 commented on README.md:
```diff
@@ -1,24 +1,63 @@
+### buyテーブル
+|column     |types    |Options                       |
+|-----------|---------|------------------------------|
+|item       |reference|null: false, foreign_key: true|
+|user       |reference|null: false, foreign_key: true|
```
```
|item       |reference|null: false, foreign_key: true|
|user       |reference|null: false, foreign_key: true|
```

こちらも以下のように修正しましょう！

```
|item       |references|foreign_key: true|
|user       |references|foreign_key: true|
```

sgwr-0113 commented on README.md:
```diff
@@ -1,24 +1,63 @@
+|addresses     |string   |null: false                   |
+|building      |string   |                              |
+|phone_number  |string   |null: false                   |
+|buy           |reference|null: false, foreign_key: true|
```
```
|buy           |reference|null: false, foreign_key: true|
```

こちらも修正しましょう！

sgwr-0113 reviewed (commented):
レビュー依頼ありがとうございます！
コメントの確認をお願いします。

また、カラムを修正した**最新版のER図**の提出も合わせてお願いいたします。

だいぶ記述が整ってきましたね、もう一息です！
修正後、再度フォームにてレビュー依頼をお願いいたします。

akirausui19871117 commented:
理解不足でした・・・。御確認いただきましてありがとうございました。
ご指摘の通り修正いたしました。お手数ですが、再度御確認頂ますようお願いいたします。

https://gyazo.com/6c47ba9591f84957a3d68710d2a9afd6

akirausui19871117 committed:
修正その５

外部キーの制約に対するスペルミスの修正と、不要な記述の削除

chisesakai commented on README.md:
```diff
@@ -1,24 +1,63 @@
+belongs_to :user
+has_one :buy
+
+### buyテーブル
```
```
### buyテーブル
```
テーブル名は複数形にしましょう！

chisesakai commented on README.md:
```diff
@@ -1,24 +1,63 @@
+belongs_to :item
+has_one :destination
+
+### destinationテーブル
```
```
### destinationテーブル
```
こちらも同様に、複数形に変更しましょう。

chisesakai commented on README.md:
```diff
@@ -1,24 +1,63 @@
+### itemsテーブル
+|column          |types         |Options                     |
+|----------------|--------------|----------------------------|
+|item_name       |string        |null: false                 |
```
```
|item_name       |string        |null: false                 |
```
`item_`の部分を削除しましょう。
理由
テーブル名から分かる情報はカラム名に記載しない方が良いとされているためです。

chisesakai reviewed (commented):
レビュー依頼ありがとうございます。
修正点確認させていただきました。
#### 追加で3件コメントいたしましたので、ご確認お願いいたします。
もう完成間近ですね。あと一息、頑張っていきましょう！

また、ER図に更新されていない箇所(destinationsテーブルに外部キーを保存するカラムがない、等)が複数見受けられますので、新しいER図の添付もよろしくお願い致します。
修正頂いた後、コミットプッシュを確認いただき、再度レビュー依頼のご提出をお願い致します！

akirausui19871117 committed:
修正その６

テーブル名を複数形に変更、itemsテーブルのitem_nameをnameへ変更

akirausui19871117 commented:
再三に渡る御確認ありがとうございました！ご指摘の箇所を修正いたしました！またあわせてテーブルの名前も修正いたしました。
再度御確認お願いいたします！

akirausui19871117 commented:
https://gyazo.com/11d75500c5bbf820b33f097b21ddc59c

maimai0401 reviewed (commented):
レビュー依頼ありがとうございます。
LGTMとさせていただきます！実装お疲れ様でした。

マージ後、ローカルのmasterブランチに忘れずにプルしましょう。
その後、新しいブランチを作成し次の実装に取り組んでいただけますと幸いです。
引き続き実装頑張っていきましょう！
